File: solutions/typescript/resistor-color-duo/2/resistor-color-duo.ts
export enum COLORS {
  black,
  brown,
  red,
  orange,
  yellow,
  green,
  blue,
  violet,
  grey,
  white,
};

type Color = keyof typeof COLORS;

export function decodedValue(colors: Color[]): number {
  return 10 * COLORS[colors[0]] + COLORS[colors[1]];
}

File: solutions/python/resistor-color-duo/2/resistor_color_duo.py
COLORS = (
        'black', 
        'brown', 
        'red', 
        'orange', 
        'yellow', 
        'green', 
        'blue', 
        'violet', 
        'grey', 
        'white'
)

def value(colors: list[str]) -> int:
    return int(f'{COLORS.index(colors[0])}{COLORS.index(colors[1])}')
